src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -357,6 +358,159 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
             return expandedBytes;
         }
 
+        private static unsafe string UrlEncode(char* s, int length)
+        {
+            int numUnsafe = 0;
+            int numSpaces = 0;
+            for (int i = 0; i < length; i++)
+            {
+                char c = s[i];
+                if (c == ' ')
+                {
+                    numSpaces++;
+                }
+                else if (!IsUrlSafeChar(s[i]))
+                {
+                    numUnsafe += GetByteCount(s, i, length);
```

Comment:
We should not have our implementation for UTF8Encoding, even if it is "faster". We can live with the performance of the current implementation until such time as we are able to rev UTF8Encoding in CoreCLR.